core/src/main/scala/com/salesforce/op/evaluators/OpMultiClassificationEvaluator.scala

```diff
@@ -118,19 +129,68 @@ private[op] class OpMultiClassificationEvaluator
         thresholds = $(thresholds)
       )
 
+      val metricsTopK = calculateTopKMetrics(
+        data = dataUse.select(col(probabilityColName), col(labelColName).cast(DoubleType)).rdd.map{
+          case Row(prob: Vector, label: Double) => (prob.toArray, label)
+        },
+        topKs = $(topKs)
+      )
+
       val metrics = MultiClassificationMetrics(
         Precision = precision,
         Recall = recall,
         F1 = f1,
         Error = error,
-        ThresholdMetrics = thresholdMetrics
+        ThresholdMetrics = thresholdMetrics,
+        TopKMetrics = metricsTopK
       )
 
       log.info("Evaluated metrics: {}", metrics.toString)
       metrics
     }
   }
 
+/**
+ * @param data       Input RDD consisting of (vector of score probabilities, label), where label corresponds to the
+ *                   index of the true class and the score vector consists of probabilities for each class
+ * @param topKs      Sequence of topK values to calculate multiclass classification metrics for
+ */
+  def calculateTopKMetrics(
+    data: RDD[(Array[Double], Double)],
+    topKs: Seq[Int]
+  ): MultiClassificationMetricsTopK = {
+    type Label = Int
+
+    def computeMetrics(scoresAndLabels: (Array[Double], Double)): (Array[Double], Double) = {
+      val scores: Array[Double] = scoresAndLabels._1
+      val label: Label = scoresAndLabels._2.toInt
+      val topKsAndScores: Map[Label, Array[(Double, Int)]] = topKs.map(t => t -> scores.zipWithIndex.sortBy(-_._1)
```

Comment:
I don't think this is doing what we want.

This seems to be repeating the existing topN calculation in the threshold metrics (eg. allow a "correct" to correspond to the true label class being in the top 1, top 3, or top 10 of the predicted labels).

What we want is a measure of model metrics (error, precision, recall, f1) restricted to the topK most commonly occurring labels. Eg. what's the error if we only allow the top 10 most common labels to be predicted? The topN is always 1 here still - we only care about the prediction label with the highest score, but if it's outside the topK most common labels then it wouldn't count (even if it's a correct classification).